src/System.Runtime.Analyzers/UnitTests/DoNotRaiseReservedExceptionTypesTests.cs

```diff
@@ -10,6 +10,119 @@ namespace System.Runtime.Analyzers.UnitTests
 {
     public class DoNotRaiseReservedExceptionTypesTests : DiagnosticAnalyzerTestBase
     {
+        [Fact]
+        public void CreateSystemNotImplementedException()
+        {
+            VerifyCSharp(@"
```

Comment:
I personally prefer this approach when possible, what do you guys think?

@srivatsn @lgolding @michaelcfanning